pkgs/top-level/python-packages.nix

```diff
@@ -10042,6 +10048,8 @@ in {
 
   jupyter_core = callPackage ../development/python-modules/jupyter_core { };
 
+  jupyterhub = callPackage ../development/python-modules/jupyterhub { };
```

Comment:
I don't think this should be in pythonPackages (or python-modules). Since it provides a binary to run, I think this should be a separate application in top-level

Reply:
Yes and no, its a bit tricky. The jupyterhub and the notebook are applications, and could/should be outside of `python-packages.nix`, but notebook needs a kernel and that one needs to be inside `python-packages.nix`. Doing it that way requires a bit more setup to get e.g. a Python 3 kernel with a Python 2 notebook, so this is a more pragmatic solution. In the long-term I think we should go in that direction, providing a function that allows you to add kernels to your notebook.